src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpVariableDeclarationNode.cs

```diff
@@ -17,7 +17,7 @@ internal HttpVariableDeclarationNode(SourceText sourceText, HttpSyntaxTree? synt
     {
     }
 
-    public string VariableName => ChildTokens.Where(t => t.Kind == HttpTokenKind.Word).Select(t => t.Text).FirstOrDefault() ?? string.Empty;
+    public string VariableName => string.Join("", ChildTokens.Where(t => t.Kind == HttpTokenKind.Word || (t.Kind == HttpTokenKind.Punctuation && t.Text is "_")).Select(t => t.Text)) ?? string.Empty;
```

Comment:
Isn't this just `this.Text`?

https://github.com/dotnet/interactive/blob/main/src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpSyntaxNodeOrToken.cs#L39

Reply:
Ah looks like the above node is for the entire declaration and not just the variable name. I think it may be better to have a dedicated sub-node for the variable name (which could potentially also be used to parse the variable references that appear inside `{{}}` expressions).

Reply:
We could create a sub-node but I think the main utility would just be getting the value of the text and reconfiguring the parser to use this new node and I'm not sure that it is necessarily cleaner and worth the rework

Reply:
I may be wrong, but I think we'll need to parse variable names once we start interpreting expressions and I think we may need to start doing that for the named requests feature (to implement support for [this syntax](https://marketplace.visualstudio.com/items?itemName=humao.rest-client#:~:text=%7B%7BrequestName.(response%7Crequest).(body%7Cheaders).(*%7CJSONPath%7CXPath%7CHeader%20Name)%7D%7D) for dotting into the named requests).

We may be able to sidestep this by using a separate / simpler (regex-based?) parser just for parsing the expression contents. Not sure if that is the path we will decide to go down though...

In any case, your call to keep this simple for now seems reasonable considering that we don't need to parse variable names in other places yet.

Reply:
I don't think a regex is the right path. The different approaches within the same parser will make it harder to understand, and the regex isn't going to support language services as well.

Reply:
+1 I had the same thoughts. Having a dedicated sub node could also make the implementation of features like go to definition / find all references simpler / more natural.